### PR TITLE
Add option to use Post Featured Image with consistent height 

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
This PR adds support for setting consistent height for the Post Featured Image block, using css `background-image` property. I think this is a very important thing for Post Featured Image to support, so as to enable basic/common designs mostly used through `Query` block.

I intentionally handled this specifically in this block by providing minimum functionality which provides value ( IMO :) ), as there are discussions for creating a generic `background` control to handle featured images, along with other options (https://github.com/WordPress/gutenberg/issues/24660).

## How has this been tested?
Manually

## Screenshots 
useAsCover

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/post-featured-image/style.scss [hunk 1]
```diff
@@ -6,4 +6,17 @@
 		max-width: 100%;
 		height: auto;
 	}
+	&.is-used-as-cover {
+		position: relative;
+		min-height: 300px;
+		background-position: center center;
+		background-size: cover;
+		a {
```
Comment (ntsekouras) [inline packages/block-library/src/post-featured-image/style.scss:14]: @jasmussen is there a better way to achieve full height/width link with no content?

**Comment by aristath** [comment]
I'm a little confused... Is the purpose of this PR to allow consistent height for all featured image blocks when used in a grid?
The wording "cover" here is basically what confused me, I was expecting to be able to add inner blocks - just like in the cover block - which is not the case here (I saw the title of the PR and thought _"niiiice!! I'll be able to add the post-title on top of the image"_)
If the sole purpose is to have all featured images use a consistent height, then perhaps the wording should change to avoid confusion with the cover block?

**Comment by ntsekouras** [comment]
>If the sole purpose is to have all featured images use a consistent height, then perhaps the wording should change to avoid confusion with the cover block?

Yes. I had the css `background-size:cover` in mind :). I'll change the title/description.

**Comment by jasmussen** [comment]
Yes, I'd echo Ari, the use of "Cover" is a little misleading. I understand this to be that you are using the `background-size: cover;` on a div with a background, instead of inserting an `img` as it is now. 

I do agree, this is a nice feature for the Featured Image block to have. But having just worked on #27331, one thing that became clear to me from that effort was that we have a lot of very very different UI across sidebars, and it's making it very inconcistent and confusing. 

So my first question is: how urgent is the need for this feature (perhaps @kjellr who's working on block themes can help answer)? If it's very urgent, we can probably make some small tweaks and land this as a small solution.

If it's _not_ urgent, I think we might want to wait for some of the efforts to address #27331 to land, so we can make a consistent UI for this. Because what it boils down to is two things:

1. A toggle to choose between outputting an IMG, vs. outputting a div with a background image.
2. Controls to adjust the properties of the background image.

2 is where it gets really tricky — how deep do we go? Here's a mockup of the controls for the Cover block in the new layout:

Cover Sidebar

- Do we allow fixed background?
- Do we allow arbitrary positions or is it always centered?
- Should you be able to choose between cover/contain? How about repeat/no-repeat?

The thing is, ideally the UI here should be identical between any block that allows background images, give or take controls that get hidden on a per-block basis if they don't make sense. And for every new variation of this UI we create, we have to go back and fix up the old ones once we finally do get to a point. So while I deeply appreciate your work here, the thing is, I can see this UI being used in many places that allow backgrounds, and it might be nice to get this UI super duper right and consistent for the cover _block_, before we expand it further.

CC: @ItsJonQ @mtias

**Comment by kjellr** [comment]
> So my first question is: how urgent is the need for this feature (perhaps @kjellr who's working on block themes can help answer)? If it's very urgent, we can probably make some small tweaks and land this as a small solution.

I wouldn't describe it as urgent, though I do think this sort of thing should make its way in eventually.

File: packages/block-library/src/post-featured-image/style.scss (hunk 1 above)
Comment (scruffian) [reply packages/block-library/src/post-featured-image/style.scss:14]: would it make more sense to just wrap the image in an a tag?

**Comment by jasmussen** [comment]
A simpler if perhaps interim alternative is to use object-fit, which the gallery block does at the moment.

https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit

**Comment by ntsekouras** [comment]
>A simpler if perhaps interim alternative is to use object-fit, which the gallery block does at the moment.

Yeah, I know. I saw that we already use this in a couple of places in GB, but since it doesn't have full browser support (IE) I didn't go for it.

Anyway, it seems from the discussions that this PR will not go through and we'll wait for the new designs/implementations.

**Comment by jasmussen** [comment]
While we still support IE11, everything beyond that supports [object fit](https://caniuse.com/?search=object-fit), and I would suggest that given the feature image will still _work_ in IE11, just not cover the area, that it's a decent solution that should be somewhat simpler to implement, with this toggle in the sidebar:

Screenshot 2020-12-08 at 09 09 11

But okay to wait if you prefer the background approach.

**Comment by ntsekouras** [comment]
> But okay to wait if you prefer the background approach.

It's not about the specific approach - using `object-fit` would make things simpler code-wise.. Still this will have to use the same two controls in `InspectorControls`. I understand that the proposition is to wait for the new components design/implementation. 

I'll wait a day or two for any other comments and will close the PR..

**Comment by jasmussen** [comment]
I've added a label to get some additional thoughts.

**Comment by scruffian** [comment]
It could be interesting to work on the link aspect of this separately, so that when we are ready for this the link is already in place.

**Comment by ntsekouras** [comment]
>It could be interesting to work on the link aspect of this separately, so that when we are ready for this the link is already in place.

You mean making the Post Featured Image a link? Because it's implemented already on master.

**Comment by scruffian** [comment]
Ah OK, I wasn't aware of that. I guess the part that seemed challenging in the PR was ensuring that the image was wrapped in the a tag, but I don't know how that's working on master...

**Comment by kjellr** [comment]
I opened an issue here to continue tracking this feature: https://github.com/WordPress/gutenberg/issues/27620.

**Pull request closed without merging**